src/MapViewController.swift

```diff
@@ -128,6 +141,49 @@ public class MapViewController: TGMapViewController, LocationManagerDelegate, TG
     }
   }
 
+  public func display(route: OTRRoutingResult) throws {
+    //TODO: We eventually should support N number of routes.
+    if let routeMarker = currentRouteMarker {
+      //We don't throw if the remove fails here because we want to silently replace the route
+      currentRouteMarker = nil
+      markerRemove(routeMarker)
+    }
+    let routeLeg = route.legs[0]
```

Comment:
You can see here in fact on the road Android also only cares about `leg[0]` currently.

https://github.com/mapzen/on-the-road_android/blob/master/library/src/main/java/com/mapzen/valhalla/Route.kt#L69-L70

Let's punt for now and make an issue to add full support for waypoints later.